src/mpid/ch4/shm/ipc/src/ipc_p2p.h

```diff
@@ -52,7 +52,7 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_IPCI_send_contig_lmt(const void *buf, MPI_Ain
 
     slmt_req_hdr->src_lrank = MPIR_Process.local_rank;
     slmt_req_hdr->data_sz = data_sz;
-    slmt_req_hdr->sreq_ptr = (uint64_t) sreq;
+    slmt_req_hdr->sreq_ptr = (uintptr_t) sreq;
```

Comment:
This struct member should probably be converted to `MPIR_Request`.

Reply:
This is part of active message header. I think one of the reason to use an integer type is to prevent confusion or accidentally dereferencing it on the target side.

Reply:
We do this throughout ch4, and sometimes the struct is meant to be dereferenced. ch3 uses `MPI_Request` handles, which is actually better, but I haven't had time to convert every place in ch4 yet.

Reply:
I still think using opaque integer type makes sense here. That said, there is probably much refactoring can be done for the AM code that probably will make current argument obsolete.

Reply:
I don't see how refactoring is going to change anything in this instance. We need to pass a reference to the request back and forth during rendezvous. This is well established protocol in both ch3 and ch4.

Reply:
I am a bit lost on what is holding up here. Are you insisting on changing the struct member type to `MPIR_Request *`? I can do that, and all the other similar places to be consistent. I still think using opaque integer type makes sense here, but it is a very minor preference and I don't think it's worth defending 😄 .

Reply:
Yes, we should use `MPIR_Request *` like the rest of ch4 message headers.

Reply:
To clarify, by refactoring I was referring to that we can make the the header struct a very local struct, then it will be apparent that there is very small impact on updating the struct, and we probably will not overthink about `uintptr_t` vs `void *` here since we can just look at the code where it is used and determine what makes more sense. Right now this is not obvious, and we are obviously dragging contexts from all over (even ch3). So the refactoring may not change the fact what needs be done, it will make it context clear and thus void the need for argument.

Anyway, if my clarification makes it more confusing, let's just ignore it.

Reply:
> 
> 
> Yes, we should use `MPIR_Request *` like the rest of ch4 message headers.

I'll add a commit.

Reply:
done. I vaguely recall we had more places using opaque int types; but apparently not.